components/input/__tests__/Search.test.tsx

```
@@ -413,4 +414,26 @@ describe('Input.Search', () => {
+
+    const button = container.querySelector<HTMLButtonElement>('.ant-input-search-button')!;
+
+    // search.ts sets minHeight on the small search button to ensure it aligns
+    // with the input height. In compact algorithm, controlHeightSM is 21px but
+    // the calculated smallButtonHeight should be >= 22px (to cover the input's
+    // font line box + borders). Verify the computed min-height is set and is
+    // greater than the raw controlHeightSM (21px).
+    const buttonMinHeight = getComputedStyle(button).minHeight;
+    expect(buttonMinHeight).not.toBe('0px');
+
+    const minHeightValue = parseFloat(buttonMinHeight);
```

ESLint

[error] Prefer `Number.parseFloat` over `parseFloat`.

(unicorn/prefer-number-properties)